lib/srv/desktop/rdp/rdpclient/src/lib.rs

```diff
@@ -734,14 +808,6 @@ pub unsafe extern "C" fn free_rdp(client_ptr: *mut Client) {
     drop(Client::from_raw(client_ptr))
 }
 
-/// # Safety
-///
-/// The passed pointer must point to a C-style string allocated by Rust.
-#[no_mangle]
-pub unsafe extern "C" fn free_rust_string(s: *mut c_char) {
```

Comment:
This was unused?

Reply:
I think this might have been unused since we fixed the memory leak some time ago by passing pointers.

Reply:
Yes, this was a holdover from when we were passing strings back and forth between Rust and Go as error messages. 

As I'm writing things now, passing a string from Rust to Go looks like:
```rust
match CString::new("some string") {
    Ok(c_string) => {
        // Convert C compatible string into raw pointer for passing into Go
        let c_string = c_string.into_raw();
        unsafe {
            // pass_to_go should copy the c_string into a Go string if it needs it beyond its call
            pass_to_go(c_string)
            // Retake pointer to free memory
            let _ = CString::from_raw(c_string);
        }
        return Ok(());
    }
    Err(_) => {
        // error
    }
}
```

And passing a string from Go to Rust looks like
```go
cString := C.CString("some string")
defer C.free(unsafe.Pointer(cString))
// pass_to_rust should copy the cString if it needs it beyond this call
C.pass_to_rust(cString)
```

Essentially, the C-String's creator is always responsible for freeing the memory, and the receiving function is always responsible for copying the C-String into a native string before returning.

Reply:
Yes, as I noted in another comment, you can skip the raw conversion and back when calling into Go from Rust and just do:

```rust
unsafe {
  let ptr = c_string.as_ptr() as *mut c_char;
  pass_to_go(ptr);
  // c_string is freed since we never gave away ownership
}
```

Reply:
Is there a reason I want to use `*mut c_char` or can I just change those to `*const c_char`?

Reply:
Last I tried passing const pointers in cgo I did not have a great time, it freaked out hard but do give it a try @ibeckermayer